### PR TITLE
Update usage messages from potential supplement modules

### PR DESCRIPTION
**Pull request by PaulWessel**
As per #5341.  Lots of minor issues fixed such as mismatch between doc and usage, usage synopsis out of order, spelling issues, etc.  I would like @joa-quim to have a look at "his" modules to make sure the messages are correct.  Also, there seems to be modifiers listed for **grdgravmag3d** that are not implemented - please check.

### CODE REVIEW

**Comment by joa-quim** [comment]
Possibly, but which ones did you suspect not being implemented? **-H** is the longest one but it seems they are all parsed.

**Comment by PaulWessel** [comment]
Well, I am seeing comments like

`if (Ctrl->H.koningsberg != 0) {		/* This option is not currently implemented */`

`Ctrl->H.incfile = strdup(&opt->arg[2]);    /* Inclination grid (NOT IMPLEMENTED YET) */`

`Ctrl->H.decfile = strdup(&opt->arg[2]);        /* Declination grid (NOT IMPLEMENTED YET) */`

and the usage message says

`GMT_Message (API, GMT_TIME_NONE, "\t     For a variable inclination and declination use IGRF. Set any of -H+i|g|r|f|n to do that.\n");`

but there is no explanation for what these modifiers are either in usage or the documentation.  Perhaps you can clean up those things so that there is a match between usage, documentation, and actual implemented capability?

File: src/potential/gmtflexure.c [hunk 1]
```diff
@@ -297,43 +297,57 @@ static int parse (struct GMT_CTRL *GMT, struct GMTFLEXURE_CTRL *Ctrl, struct GMT
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s -D<rhom>/<rhol>[/<rhoi>]/<rhow> -E<te> -Q<loadinfo> [-A[l|r]<bc>[/<args>]]\n", name);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-C[p|y]<value] [-F<force>] [-L] [-S] [-T<wpre>] [%s] [-W<w0>] [-Z<zm>]\n\t[%s] [%s] [%s]\n\t[%s] [%s] [%s]\n\n",
-		GMT_V_OPT, GMT_e_OPT, GMT_e_OPT, GMT_h_OPT, GMT_i_OPT, GMT_o_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s -D<rhom>/<rhol>[/<rhoi>]/<rhow> -E<Te>[k]|<D>|<file> -Qn|q|t[<args>] [-A[l|r]<bc>[/<args>]] "
+		"[-C[p|y]<value>] [-F<force>] [-L] [-M[x][z]] [-S] [-T<wpre>] [%s] [-W<w0>[k]] [-Z<zm>[k]] [%s] [%s] [%s] [%s] [%s] [%s] [%s]\n",
```
Comment (meghanrjones) [inline src/potential/gmtflexure.c:301]:
```suggestion
		"[-Cp|y<value>] [-F<force>] [-L] [-M[x][z]] [-S] [-T<wpre>] [%s] [-W<w0>[k]] [-Z<zm>[k]] [%s] [%s] [%s] [%s] [%s] [%s] [%s]\n",
```
Directives are required according to html docs

File: src/potential/gmtflexure.c [hunk 2]
```diff
@@ -297,43 +297,57 @@ static int parse (struct GMT_CTRL *GMT, struct GMTFLEXURE_CTRL *Ctrl, struct GMT
...
+	GMT_Usage (API, 1, "\n-A[l|r]<bc>[/<args>]");
+	GMT_Usage (API, -2, "Specify boundary conditions at the ends of the profile. "
+		"Use -Al and -Ar for separate conditions at the left and right end, respectively [Same conditions]. "
+		"Four types of BC's are recognized (here, w = w(x) is the deflection):");
+	GMT_Usage (API, 3, "0: \"Infinity\" condition, w' = w'' = 0.");
+	GMT_Usage (API, 3, "1: \"Periodic\" condition, w' = w''' = 0.");
+	GMT_Usage (API, 3, "2: \"Clamped\" condition, append <w0> so w at end = <w0> [0], w' = 0.");
+	GMT_Usage (API, 3, "3: \"Free\" condition, append m/f as (m)oment and (f)orce at end [0/0].");
+	GMT_Usage (API, -2, "Default BCs are -A0.  Use SI units for any optional arguments.");
+	GMT_Usage (API, 1, "\n-C[p|y]<value>");
```
Comment (meghanrjones) [inline src/potential/gmtflexure.c:330]:
```suggestion
	GMT_Usage (API, 1, "\n-Cp|y<value>");
```

File: src/potential/gmtgravmag3d.c [hunk 3]
```diff
@@ -176,54 +176,70 @@ GMT_LOCAL int check_triang_cw (struct GMTGRAVMAG3D_CTRL *Ctrl, unsigned int n, u
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s xyz_file -Tv<vert_file> | -Tr|s<raw_file> OR -M+s<par> [-C<density>] [-G<outgrid>]\n", name);
-	GMT_Message (API, GMT_TIME_NONE, "\t[%s] [%s] [-E<thick>] [-F<xy_file>] [-L<z_observation>]\n", GMT_I_OPT, GMT_Rgeo_OPT);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-H<f_dec>/<f_dip>/<m_int></m_dec>/<m_dip>] [-S<radius>]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t[-Z<level>] [%s] [-fg] [%s] [%s]\n\n", GMT_V_OPT, GMT_r_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [<xyz_file>] -Tv<vert_file> | -Tr|s<raw_file> OR -M+s<body>/<pars> [-C<density>] [-E<thickness>] "
```
Comment (meghanrjones) [inline src/potential/gmtgravmag3d.c:179]:
```suggestion
	GMT_Usage (API, 0, "usage: %s [<xyz_file>] -Tv<vert_file> | -Tr|s<raw_file> | -M+s<body>/<pars> [-C<density>] [-E<thickness>] "
```

File: src/potential/grdflexure.c [hunk 4]
```diff
@@ -753,49 +753,65 @@ static int parse (struct GMT_CTRL *GMT, struct GRDFLEXURE_CTRL *Ctrl, struct GMT
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s <topogrid> -D<rhom>/<rhol>[/<rhoi>]/<rhow> -E[<te>[/<te2>]] -G<outgrid> [-A<Nx/Ny/Nxy>] [-C[p|y]<value] [-F<nu_a>[/<h_a>/<nu_m>]]\n", name);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-L<list>] [-M<tm>] [-N%s] [-Q] [-S<beta>] [-T<t0>[/<t1>/<dt>]|<file>|<n>[+l]]]\n\t[%s] [-W<wd>[k]] [-Z<zm>[k]] [-fg] [%s]\n\n", GMT_FFT_OPT, GMT_V_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s <topogrid> -D<rhom>/<rhol>[/<rhoi>]/<rhow> -E[<te>[k][/<te2>[k]]] -G<outgrid> [-A<Nx/Ny/Nxy>] [-C[p|y]<value] [-F<nu_a>[/<h_a>[k]/<nu_m>]] "
```
Comment (meghanrjones) [inline src/potential/grdflexure.c:756]:
```suggestion
	GMT_Usage (API, 0, "usage: %s <topogrid> -D<rhom>/<rhol>[/<rhoi>]/<rhow> -E[<te>[k][/<te2>[k]]] -G<outgrid> [-A<Nx/Ny/Nxy>] [-Cp|y<value] [-F<nu_a>[/<h_a>[k]/<nu_m>]] "
```

File: src/potential/grdflexure.c [hunk 5]
```diff
@@ -753,49 +753,65 @@ static int parse (struct GMT_CTRL *GMT, struct GRDFLEXURE_CTRL *Ctrl, struct GMT
...
-	GMT_Message (API, GMT_TIME_NONE, "\t-C use -Cy<Young> or -Cp<poisson> to change Young's modulus [%g] or Poisson's ratio [%g].\n", YOUNGS_MODULUS, POISSONS_RATIO);
-	GMT_Message (API, GMT_TIME_NONE, "\t-F Sets upper mantle viscosity, and optionally its thickness and lower mantle viscosity.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   Viscosity units in Pa s; thickness in meter (append k for km).\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-L Give filename for output table with names of all grids (and model times) produced.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t   If no filename is given then we write the list to standard output.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-M Set Maxwell time for viscoelastic flexure (in years; append k for kyr and M for Myr).\n");
+	GMT_Usage (API, 1, "\n-A<Nx/Ny/Nxy>");
+	GMT_Usage (API, -2, "Set in-plane force components <Nx>, <Ny> and shear force <Nxy> in Pa*m [isotropic deformation]. "
+		"Negative values mean compression, positive values mean extensional forces.");
+	GMT_Usage (API, 1, "\n-C[p|y]<value>");
```
Comment (meghanrjones) [inline src/potential/grdflexure.c:785]:
```suggestion
	GMT_Usage (API, 1, "\n-Cp|y<value>");
```

File: src/potential/grdgravmag3d.c [hunk 6]
```diff
@@ -191,54 +191,64 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GRDGRAVMAG3D_CTRL *C) {	/* D
...
-	GMT_Message (API, GMT_TIME_NONE, "\t-C Sets body density in SI.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-F Passes file with locations where anomaly is going to be computed.\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t-G Sets name of the output grdfile.\n");
+	GMT_Message (API, GMT_TIME_NONE, "  REQUIRED ARGUMENTS:\n");
+	GMT_Usage (API, 1, "\n<grdfile_top> is the grdfile whose gravity effect is to be computed. "
+		"If <grdfile_bot> is also provided then the gravity/magnetic effect of the "
+		"volume between them is computed.");
+	GMT_Usage (API, 1, "\n-C<density>");
+	GMT_Usage (API, -2, "Set body density in SI.");
+	GMT_Usage (API, 1, "\n--F<xy_file>");
```
Comment (meghanrjones) [inline src/potential/grdgravmag3d.c:207]:
```suggestion
	GMT_Usage (API, 1, "\n-F<xy_file>");
```

File: src/potential/grdseamount.c [hunk 7]
```diff
@@ -159,53 +159,72 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GRDSEAMOUNT_CTRL *C) {	/* De
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [infile(s)] -G<outgrid> %s\n\t%s [-A[<out>/<in>]] [-C[c|d|g|o|p]] [-D%s]\n", name, GMT_I_OPT, GMT_Rgeo_OPT, GMT_LEN_UNITS2_DISPLAY);
-	GMT_Message (API, GMT_TIME_NONE, "\t[-E] [-F[<flattening>]] [-L[<hcut>]] [-M[<list>]] [-N<norm>] [-Q<bmode><fmode>[+d]] [-S<r_scale>]\n");
-	GMT_Message (API, GMT_TIME_NONE, "\t[-T<t0>[/<t1>/<dt>|<file>|<n>[+l]]] [-Z<base>] [%s] [%s] [%s] [%s]\n\t[%s] [%s]\n\t[%s] [%s]\n\n",
-		GMT_bi_OPT, GMT_di_OPT, GMT_e_OPT, GMT_f_OPT, GMT_h_OPT, GMT_i_OPT, GMT_r_OPT, GMT_PAR_OPT);
+	GMT_Usage (API, 0, "usage: %s [<table>] -G<outgrid> %s %s [-A[<out>/<in>]] [-C[c|d|g|o|p]] [-D%s] "
+		"[-E] [-F[<flattening>]] [-L[<hcut>]] [-M[<list>]] [-N<norm>] [-Q<bmode><fmode>[+d]] [-S<r_scale>] "
+		"[-T<t0>[/<t1>/<dt>|<file>|<n>[+l]]] [-Z<base>] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s]\n",
```
Comment (meghanrjones) [inline src/potential/grdseamount.c:164]:
```suggestion
		"[-T<t0>[/<t1>/<dt>|<file>|<n>[+l]]] [%s] [-Z<base>] [%s] [%s] [%s] [%s] [%s] [%s] [%s] [%s]\n",
```

File: src/potential/grdseamount.c [hunk 8]
```diff
@@ -159,53 +159,72 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GRDSEAMOUNT_CTRL *C) {	/* De
...
+		name, GMT_I_OPT, GMT_Rgeo_OPT, GMT_LEN_UNITS2_DISPLAY, GMT_bi_OPT, GMT_di_OPT, GMT_e_OPT,
```
Comment (meghanrjones) [inline src/potential/grdseamount.c:165]:
```suggestion
		name, GMT_I_OPT, GMT_Rgeo_OPT, GMT_V_OPT, GMT_LEN_UNITS2_DISPLAY, GMT_bi_OPT, GMT_di_OPT, GMT_e_OPT,
```

**Review by meghanrjones — APPROVED** [review]
Formatting looks good.

**Comment by joa-quim** [comment]
> if (Ctrl->H.koningsberg != 0) { /* This option is not currently implemented */

Yes not implemented as ``Ctrl->H.koningsberg`` is initialized to 0 and never changed. But nowhere is a reference to this option that clearly is waiting for a way to select it ... and that I recall to what was the intended use.

> Ctrl->H.incfile = strdup(&opt->arg[2]); /* Inclination grid (NOT IMPLEMENTED YET) */
Ctrl->H.decfile = strdup(&opt->arg[2]); /* Declination grid (NOT IMPLEMENTED YET) */

Yes, not implemented but also never mentioned in the docs that one could use grids with variable declination and inclination.

> -H+i|g|r|f|n

Made PR #5531 to explain when they may be of use.

**Pull request merged**